src/olympia/discovery/admin.py

```diff
@@ -38,5 +42,26 @@ def formfield_for_foreignkey(self, db_field, request=None, **kwargs):
         return super(DiscoveryItemAdmin, self).formfield_for_foreignkey(
             db_field, request, **kwargs)
 
+    def build_preview(self, obj):
+        return format_html(
+            u'<br/><h2 class="heading">{}</h2>'
+            u'<div class="editorial-description">{}</div>',
+            mark_safe(obj.heading),
+            mark_safe(obj.description))
+
+    def previews(self, obj):
+        translations = []
+        with translation.override('en-US'):
+            translations.append(self.build_preview(obj))
+        with translation.override('de'):
+            translations.append(self.build_preview(obj))
+        with translation.override('es'):
+            translations.append(self.build_preview(obj))
+        with translation.override('fr'):
+            translations.append(self.build_preview(obj))
+        with translation.override('it'):
+            translations.append(self.build_preview(obj))
+        return format_html(u''.join(translations))
```

Comment:
Looking at this in the admin…

![screenshot from 2018-07-09 18-03-21](https://user-images.githubusercontent.com/139033/42462048-6cd20a60-83a2-11e8-9f2b-41f2d436f760.png)

It feels like it would be super helpful to add the information about what language this specific block is into the preview. It can be obvious for languages someone knows but as soon as we add Chinese in addition to Japanese or other arabic / rtl languages it will be crazy confusing.

I haven't tested this in RTL yet but I actually wonder how it'd look like :thinking:  and what may be needed. Although this is most certainly just used for a "small peek how it looks like" so that these details may not be too important.